src/Bicep.Decompiler/BicepDecompiler.cs

```diff
@@ -111,12 +131,19 @@ public async Task<DecompileResult> Decompile(Uri entryJsonUri, Uri entryBicepUri
                 continue;
             }
 
-            filesToSave[fileUri] = PrettyPrinter.PrintProgram(bicepFile.ProgramSyntax, new PrettyPrintOptions(NewlineOption.LF, IndentKindOption.Space, 2, false));
+            filesToSave[fileUri] = PrettyPrinter.PrintProgram(bicepFile.ProgramSyntax, GetPrettyPrintOptions());
         }
 
         return filesToSave.ToImmutableDictionary();
     }
 
+    private static string PrintSyntax(SyntaxBase syntax)
+    {
+        return PrettyPrinter.PrintSyntax(syntax, GetPrettyPrintOptions());
```

Comment:
Is there a way we can take into account the current indent level here?
For example, pasting here (`|`):
```bicep
var bar = {
  def: |
}
```
Currently results in:
```bicep
var bar = {
  def: {
  baz: 'ghi'
}
}
```
Whereas ideally it should be:
```bicep
var bar = {
  def: {
    baz: 'ghi'
  }
}
```

Reply:
`FormatHelper.ToFormattedTextWithNewIndentation()` is the method I used for the single -> multi line object code fix provider - might be helpful here.

Reply:
With the current design, it would cause another undo level to be created, which I think would be undesirable, and the workaround is simply to format document. 

Moved it to https://github.com/Azure/bicep/issues/9598.  To do so I need to move more of the code into the language server, which is a good idea anyway, and also probably necessary for #9509.